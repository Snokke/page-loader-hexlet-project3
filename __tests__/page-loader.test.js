import nock from 'nock';
import axios from 'axios';
import fs from 'fs';
import url from 'url';
import os from 'os';
import path from 'path';
import httpAdapter from 'axios/lib/adapters/http';
import loader from '../src';

axios.defaults.adapter = httpAdapter;
nock.disableNetConnect();

test('Download html page', async () => {
  const host = 'https://hexlet.io';
  const pathname = '/courses';
  const body = 'html body';
  const filePath = url.format({ host, pathname });

  nock(host).get(pathname).reply(200, body);

  const tempDir = await fs.promises.mkdtemp(path.join(os.tmpdir()));
  await loader(filePath, tempDir);
  const fileName = await fs.promises.readdir(tempDir);
  const tempFilePath = path.resolve(tempDir, fileName[0]);
  const data = await fs.promises.readFile(tempFilePath);

  expect(data.toString('utf8')).toBe(body);
});
