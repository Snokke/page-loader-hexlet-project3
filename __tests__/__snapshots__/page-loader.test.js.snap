// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error: wrong dir 1`] = `"Wrong path: /wrong/path/hexlet-io-courses.html"`;

exports[`Error: wrong url 1`] = `"Request failed with status code 404"`;
